playbooks/pvc_renew_certs.yml

```diff
@@ -0,0 +1,131 @@
+---
+
+# Copyright 2024 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Pre - Initialization of local working directories
+  hosts: localhost
+  connection: local
+  tasks:
+    - name: Create temporary build directory
+      ansible.builtin.tempfile:
+        state: directory
+        prefix: pvc_tls_
+      register: __pvc_tls_tempdir
+
+    - debug:
+        var: __pvc_tls_tempdir
+
+    - name: Create a directory for csrs and signed certs
+      ansible.builtin.file:
+        path: "{{ [__pvc_tls_tempdir.path, item] | path_join }}"
+        state: directory
+        mode: '0755'
+      loop:
+        - csrs
+        - certs
+        - ca_certs
+
+- name: Play 1 - Generate CSR on each host
+  hosts: "{{ target | default('cluster') }}"
+  become: yes
+  gather_facts: yes
+  tasks:
+
+  - name: Call tls_generate_csr role
+    ansible.builtin.import_role:
+      name: cloudera.exe.tls_generate_csr
+    vars:
+      local_csrs_dir: "{{ (hostvars['localhost']['__pvc_tls_tempdir']['path'], 'csrs') | path_join }}"
+
+- name: Get the list of CSRs to sign
+  hosts: localhost
+  connection: local
+  tasks:
+    - name: "Set fact for all CSRs in {{ local_csrs_dir }}"
+      ansible.builtin.find:
+        paths: "{{ local_csrs_dir }}"
+        file_type: file
+      register: __csrs_to_sign
+      vars:
+        local_csrs_dir: "{{ (hostvars['localhost']['__pvc_tls_tempdir']['path'], 'csrs') | path_join }}"
+
+    - name: Set fact for csrs to sign
+      ansible.builtin.set_fact:
+        local_csrs_to_sign: "{{ __csrs_to_sign.files | json_query('[*].path') | flatten }}"
+
+- name: Play 2 - Sign the CSR
+  hosts: ca_server
```

Comment:
Correct, the flow for renewing FreeIPA certs will be slightly different as it can happen on the cluster host with the step outlined here: https://www.freeipa.org/page/Certmonger#manually-renew-a-certificate.